### PR TITLE
Create UI to watch Bookies and Ledgers state

### PR DESCRIPTION
**Pull request by mino181295**
Implementations in this commits:

- Created methods to access Bookies info, Ledgers, and Ledgers for a spcific Bookie (based on hostname)
- Created the Frontend project with the navigation through Bookie page, Bookie ledgers page and all Ledgers page.
- Exposed the API for /bookie and /ledger 

**Note**: this pull request lacks of optimizations

### CODE REVIEW

File: src/main/java/org/bookkeepervisualmanager/bookkeeper/BookkeeperManager.java [hunk 1]
```diff
@@ -117,14 +128,66 @@ public BookKeeper getBookkeeper() {
...
+                    ledgerMetadata.set(metadata.getValue());
+                }
+            }).join();
+            return ledgerMetadata.get();
+        } catch (Throwable e) {
+            throw new BookkeeperException(e);
+        }
+    }
+
+    public Map<BookieSocketAddress, BookieInfo> getAvailableBookies() throws BookkeeperException {
```
Comment (eolivelli) [inline src/main/java/org/bookkeepervisualmanager/bookkeeper/BookkeeperManager.java]: getBookieInfo ?

File: src/main/java/org/bookkepervisualmanager/api/resources/LedgersResource.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        for (Entry<Long, LedgerMetadata> currentLedger : allLedgers.entrySet()) {
+            LedgerBean b = new LedgerBean();
+
+            long lid = currentLedger.getKey();
+            b.setId(lid);
+
+            LedgerMetadata ledgerMetadata = currentLedger.getValue();
+            Map<String, byte[]> customMetadata = ledgerMetadata.getCustomMetadata();
+            for (Entry<String, byte[]> currentCustomMetadata : customMetadata.entrySet()) {
+                b.setMetadataValue(currentCustomMetadata.getKey(), currentCustomMetadata.getValue());
```
Comment (eolivelli) [inline src/main/java/org/bookkepervisualmanager/api/resources/LedgersResource.java]: convert byte[] > String(utf-8)

File: src/main/java/org/bookkepervisualmanager/api/resources/LedgersResource.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        for (Entry<Long, LedgerMetadata> currentLedger : forBookie.entrySet()) {
+            LedgerBean b = new LedgerBean();
+
+            long lid = currentLedger.getKey();
+            b.setId(lid);
+
+            LedgerMetadata ledgerMetadata = currentLedger.getValue();
+            Map<String, byte[]> customMetadata = ledgerMetadata.getCustomMetadata();
+            for (Entry<String, byte[]> currentCustomMetadata : customMetadata.entrySet()) {
+                b.setMetadataValue(currentCustomMetadata.getKey(), currentCustomMetadata.getValue());
```
Comment (eolivelli) [inline src/main/java/org/bookkepervisualmanager/api/resources/LedgersResource.java]: convert byte[] -> string(utf-8)

create an utility method to map LergerMetadata -> LedgerBean

File: src/main/java/org/bookkepervisualmanager/api/resources/LedgersResource.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        public Map<String, String> getMetadata() {
+            return metadata;
+        }
+
+        public void setMetadata(Map<String, String> metadata) {
+            this.metadata = metadata;
+        }
+
+        public void setMetadataValue(String key, byte[] value) throws UnsupportedEncodingException {
+            this.metadata.put(key, new String(value, "UTF-8"));
```
Comment (eolivelli) [inline src/main/java/org/bookkepervisualmanager/api/resources/LedgersResource.java]: got it !

File: src/main/java/org/bookkeepervisualmanager/bookkeeper/BookkeeperManager.java (hunk 1 above)
Comment (mino181295) [reply src/main/java/org/bookkeepervisualmanager/bookkeeper/BookkeeperManager.java]: That's better

File: src/main/java/org/bookkepervisualmanager/filters/ForceHeadersAPIRequestsFilter.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+@WebFilter(filterName = "ForceHeadersAPIRequestsFilter", urlPatterns = {"/*"})
+public class ForceHeadersAPIRequestsFilter implements Filter {
+
+    @Override
+    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletResponse resp = (HttpServletResponse) response;
+        resp.setHeader("Cache-Control", "no-cache");
+        resp.setHeader("Access-Control-Allow-Origin", "*");
```
Comment (eolivelli) [inline src/main/java/org/bookkepervisualmanager/filters/ForceHeadersAPIRequestsFilter.java]: This should not be needed.
The API will be served by the same host name as the html ui, won't it?

**Pull request merged**